test/parser_test.rb

```diff
@@ -4,13 +4,16 @@
 
 class ParserTest < Minitest::Test
 
-  def setup
-    @parser = Docurium::DocParser.new
+  def teardown
+    @parser.cleanup! if @parser
   end
 
   # e.g. parse('git2/refs.h')
   def parse(path, contents)
-    @parser.parse_file(path, [[path, contents]])
+    contents = [[path, contents]] if contents.is_a? String
```

Comment:
When is `contents` not a `String`? Is it `nil`? If we're passing other stuff we should be fixing that.

Reply:
Since I'm now managing the temporary files as part of the DocParser, it seemed easier to use this method as the main test entry point. Its "signature" is `(path, contents)` or `(path, array_of_path_contents_arrays)`, and this code converts the former form into the latter so it can be passed on.

Reply:
That feels a bit dirty, though rather ruby-ish... this should at least be documented in this method's doc